app/test_main.py

```diff
@@ -2,0 +1,23 @@
+from unittest.mock import patch
+from app.main import can_access_google_page
+
+
+def test_accessible_page() -> None:
+    with (patch("app.main.has_internet_connection", return_value=True),
+          patch("app.main.valid_google_url", return_value=True)):
+        result = can_access_google_page("http://www.google.com")
+        assert result == "Accessible"
+
+
+def test_not_accessible_no_internet() -> None:
+    with (patch("app.main.has_internet_connection", return_value=False),
+          patch("app.main.valid_google_url", return_value=True)):
+        result = can_access_google_page("http://www.google.com")
+        assert result == "Not accessible"
+
+
+def test_not_accessible_invalid_url() -> None:
+    with (patch("app.main.has_internet_connection", return_value=True),
+          patch("app.main.valid_google_url", return_valur=True)):
+        result = can_access_google_page("http://www.google.com")
+        assert result == "Not accessible"
```

Comment:
The test `test_not_accessible_invalid_url` is supposed to check for an invalid URL scenario. However, the `return_value` for `valid_google_url` is set to `True`, which contradicts the test's purpose. It should be set to `False` to properly simulate an invalid URL scenario.